tests/functional/tests.py

```diff
@@ -68,31 +74,120 @@ def read(self, amt=64*1024):
+    # Remove the first args_list element, which is always "self"
+    del args_list[0]
+    # Initialize the args , (arg: value), dictionary
+    args_dict = {value: '' for key, value in enumerate(args_list)}
+    # Create and return log output content
+    return {'name': 'minio-py',\
+            'function': meth.__name__+'('+', '.join(args_list)+')',\
+            'description': '',\
+            'args': args_dict,\
+            'duration': 0,\
+            'alert': alert,\
+            'message': None,\
+            'error': None,\
+            'status': 'PASS'}
+
+def byteify(input):
+    if isinstance(input, dict):
+        return {byteify(key): byteify(value)
+                for key, value in input.iteritems()}
+    elif isinstance(input, list):
+        return [byteify(element) for element in input]
+    elif isinstance(input, unicode):
+        return input.encode('utf-8')
+    else:
+        return input
+
+def print_json(log_output):
+    # The following code assumes log_output is a flat dictionary with
+    # the exception of args as shown above in initialize_log_output method.
+    log = {'args': {}}
+    for key, val in log_output.items():
+        # print('OUT', key, val, type(key).__name__)
+        if key == 'args':
+            for arg_key, arg_val in log_output['args'].items():
+                # print('IN', arg_key, arg_val, type(log_output['args'][arg_key]).__name__)
+                if type(log_output['args'][arg_key]).__name__ == 'str':
+                    # print('urlencoding')
+                    log['args'][urlencode(arg_key)] = urlencode(arg_val)
+                else:
+                    log['args'][arg_key] = arg_val
+
+        else:
+            log[key] = val
+    if sys.version_info[0] < 3:
+        print(byteify(json.loads(json.dumps(log))))
+    else:
+        print(json.loads(json.dumps(log)))
+
 def test_make_bucket(client):
+    start_time = time.time()
+
+    log_output = initialize_log_output(client.make_bucket)
+    log_output['description'] = 'Tests make_bucket api'
+    # Get a unique bucket_name
+    log_output['args']['bucket_name'] = bucket_name = uuid.uuid4().__str__()
+
     is_s3 = client._endpoint_url.startswith("s3.amazonaws")
-    # Get unique bucket_name, object_name.
-    bucket_name = uuid.uuid4().__str__()
-    object_name = uuid.uuid4().__str__()
-    client.make_bucket(bucket_name)
-    # Check if bucket was created properly.
-    client.bucket_exists(bucket_name)
-    # Remove buckets
-    client.remove_bucket(bucket_name)
-    ## Check if return codes a valid from server.
+    try:
+        # Create a bucket
+        client.make_bucket(bucket_name)
+        # Check if bucket was created properly
+        client.bucket_exists(bucket_name)
+        # Remove bucket
+        client.remove_bucket(bucket_name)
+    except Exception as err:
+        log_output['message'] = err
+        log_output['error'] = traceback.format_exc()
+        log_output['status'] = 'FAIL'
```

Comment:
I think we need to find a way to avoid setting 'FAIL' as a string to the status field, if we'll have set_status() method, we can create new type or two constants FAIL = 'FAIL' and SUCCESS = 'SUCCESS', that would be better to avoid hard coding it.

Reply:
Done. It'll also be touched again while refactoring and adding a Class effort.